### PR TITLE
Allow Remora Plugins to add DiscordService using only service collection.

### PR DESCRIPTION
**Pull request by AraHaan**
Some people might prefer to use Remora.Plugins for providing all of the Discord Functionality in their programs. As such with this it gives them the ability to do so.

### CODE REVIEW

**Comment by VelvetThePanda** [comment]
This PR is nonsensical; the point of `DiscordService` is to start the `DiscordGatewayClient` with a Generic Host.

**Comment by AraHaan** [comment]
And they will still have the generic host, they might just want that all Discord code be in a Remora Plugin however (which includes it using ``AddDiscordService``). It is simply a pain to have them be forced to copy the code to the IHostBuilder version of it just to use it in their plugin.

**Comment by VelvetThePanda** [comment]
Furthermore, DiscordService is a public class, so this PR is wholly unnecessary.

File: Remora.Discord.Hosting/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+[PublicAPI]
+public static class ServiceCollectionExtensions
+{
+    /// <summary>
+    /// Adds the required services for Remora Discord and a <see cref="IHostedService"/> implementation
+    /// using the service collection.
+    /// </summary>
+    /// <param name="serviceCollection">The service collection.</param>
+    /// <param name="tokenFactory">A function that retrieves the bot token.</param>
+    /// <returns>The service collection, with the services added.</returns>
```
Comment (AraHaan) [inline Remora.Discord.Hosting/Extensions/ServiceCollectionExtensions.cs:47]:
```suggestion
    /// <remarks>Only use this when the service collection is owned by an host builder.</remarks>
    /// <returns>The service collection, with the services added.</returns>
```

**Comment by AraHaan** [comment]
An alternative is to change Remora Plugins where one can define plugins that need to configure host builders (which might be a breaking change to Remora Plugins).

**Comment by VelvetThePanda** [comment]
*Or*, neither could happen, because no code needs to be changed. If you really want this behavior, you can just do it yourself; there's nothing special about Remora's methods.

**Pull request closed without merging**